modules/common/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/utils/IgluUtils.scala

```diff
@@ -243,37 +212,42 @@ object IgluUtils {
       .leftMap((sdj.schema, _))
   }
 
-  /** Check a list of SDJs and merge the Iglu errors */
+  /**
+   * Check a list of SDJs.
+   * @return All valid SDJs are in the Right while all errors are in the Left
+   */
   private def checkList[F[_]: Monad: Clock](
     client: IgluCirceClient[F],
     sdjs: List[SelfDescribingData[Json]],
     registryLookup: RegistryLookup[F]
-  ): EitherT[F, NonEmptyList[(SchemaKey, ClientError)], Unit] =
-    EitherT {
-      sdjs
-        .map(check(client, _, registryLookup).toValidatedNel)
-        .sequence
-        .map(_.sequence_.toEither)
+  ): IorT[F, NonEmptyList[(SchemaKey, ClientError)], List[SelfDescribingData[Json]]] =
+    sdjs.traverse { sdj =>
+      check(client, sdj, registryLookup)
+        .map(_ => sdj)
+        .leftMap(NonEmptyList.one)
+        .toIor
```

Comment:
I think this is wrong -- and it's my fault because my version was also wrong.

Currently, if `sdjs` contains some good and some bad, then I think this returns a `Left` not a `Both`.  I guess you want it to return a `Both`?

Furthermore... if sdjs contains only bad, then I think it would be more logical if this function can return a `Both(failures, Nil)`.  And if you can get that working, then I think you can remove the `recoverWith`s further up the page.